### ISSUE TITLE
Allow mapping of modifier keys to actions (i.e. MIDI keys or Loupedeck+ buttons)

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#1848]
This SHOULD work, but it doesn't for some strange reason - either I'm misunderstanding `hs.eventtap.event.newKeyEvent` or there's an underlying bug.

```lua
--- === plugins.core.midi.controls.midishift ===
---
--- Allows you to use a MIDI button as the SHIFT on your keyboard.

local log = require("hs.logger").new("midishift")

local plugin = {
    id = "core.midi.controls.midishift",
    group = "core",
    dependencies = {
        ["core.midi.manager"] = "manager",
    }
}

function plugin.init(deps)
    deps.manager.controls:new("shift", {
        group = "global",
        text = "MIDI: Shift Button",
        subText = "Allows a MIDI button to act like a shift button.",
        fn = function(metadata)
            --log.df("metadata: %s", hs.inspect(metadata))
            if metadata.controllerValue == 127 then
                log.df("SHIFT DOWN")
                hs.eventtap.event.newKeyEvent(hs.keycodes.map.shift, true):post()
            elseif metadata.controllerValue == 0 then
                log.df("SHIFT UP")
                hs.eventtap.event.newKeyEvent(hs.keycodes.map.shift, false):post()
            end
        end,
    })
end

return plugin
```

**Comment by latenitefilms** [comment]
Related: https://github.com/Hammerspoon/hammerspoon/issues/2104

**Comment by latenitefilms** [comment]
James Mitchell [writes](https://www.facebook.com/groups/commandpost/permalink/2574799506109078/):

> Forgive me if this has already been answered. I would like to trigger basic Modifier Keys such as SHIFT, OPTION, CONTROL from a midi controller. Is the best way to do this in KM and assign it somehow or is there a more streamlined way. Thanks in advance!

**Comment by latenitefilms** [comment]
To be addressed in #2375
